rfc/development-cycle.md

```diff
@@ -0,0 +1,180 @@
+
+Each PR should be tagged with one of three tags `semver:patch`, `semver:minor`, `semver:major` before it is merged. We could do this manually or by possibly giving a first guess [using type analysis](https://api-extractor.com/) or some commit message keywords. We should have a bot that blocks merging if one of these is not added.
+
+If you have change that is *only* a backprt, and should not appear in master, then you can target that branch as the base of this merge.
+
+### PR Backports
+Once a PR is merged into master, backport PRs should be opened against all other open branches of the corresponding increment. 
+This will likely result in some cases where backports are proposed that are not appropriate, in which case they can be closed. 
+
+1. Patch: Backport to all `{x}.{y}.x` branches and all `{x}.x.0` branches.
+2. Minor: Backport to all `{x}.x.0` branches.
+3. Major: No backports.
+
+
+### Milestones
+
+Each branch therefore always has an "active" milestone associated with it, which corresponds to the next final release
+on that branch. If we don't wish to make any more releases from a branch, we should delete it.
+
+Each PR's milestone should reflect active milestone of the branch it targets.
+
+### Changelog
+
+Each PR should add or edit a file in a directory full of files corresponding to unreleased changes, [like matplotlib does](https://matplotlib.org/devel/contributing.html#contributing-pull-requests). During a final release, these items are deleted
+from that file and compiled to a changelog file for that release. All of these files are included in the master changelog.
+
+The changelog entries should be kept in chronological order, instead of semver ordering.
+
+## Example
+
+I went back and looked at all our tags since `v1.1.0` to see if I could create a branch diagram for them (using [`gitgraph.js`](https://github.com/nicoespeon/gitgraph.js/)) if we had been using the technique I proposed above. I don't include any commits, besides releases and merges. Also, I designate still open branches by giving them each a final commit at the end. So this would be under the assumption that the only version branches open in the repo are `master`, `2.x.0`, `2.1.x`, and `1.2.x`. ([src](https://codepen.io/saulshanabrook/pen/xxwryBa?editors=1010)). 
+
+![](https://gist.githubusercontent.com/saulshanabrook/6d92df6e1872a5560674e097efd4abf3/raw/1e4f533196d24e9db9955dc8c0ba487e0633edc7/codepen----gitgraph-js-playground%2520(2).svg)
+
+## Division of labor
+
+There are many moving parts to the proposal. I thought it would be helpful to lay out what would be the responsibility of the different parties, including the bots.
+
+The "Bots" section is also a roadmap of what would need to be implemented to move forward on this.
+
+### Pull request author
+
+**Semver Label**: Make sure pull request has appropriate semver label.
+You can set it by either adding a commit with `semver:patch`, `semever:minor`, or `semver:major`, or setting the label in GitHub.
```

Comment:
There is a meesseekdev directive for this as well, outlined in our [contrib guide](https://github.com/jupyterlab/jupyterlab/blob/master/CONTRIBUTING.md#tag-issues-with-labels).